wcomponents-test-lib/src/test/java/com/github/bordertech/wcomponents/test/WCheckBoxWebElementTest.java

```diff
@@ -0,0 +1,35 @@
+package com.github.bordertech.wcomponents.test;
+
+import com.github.bordertech.wcomponents.test.components.WCheckBoxFieldUI;
+import com.github.bordertech.wcomponents.test.selenium.ByLabel;
+import com.github.bordertech.wcomponents.test.selenium.MultiBrowserRunner;
+import com.github.bordertech.wcomponents.test.selenium.SeleniumJettyTestCase;
+import com.github.bordertech.wcomponents.test.selenium.driver.SeleniumWComponentsWebDriver;
+import com.github.bordertech.wcomponents.test.selenium.element.SeleniumWCheckBoxWebElement;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(MultiBrowserRunner.class)
+public class WCheckBoxWebElementTest extends SeleniumJettyTestCase {
+
+
+	/**
+	 * Constructor...
+	 */
+	public WCheckBoxWebElementTest() {
+		super(new WCheckBoxFieldUI());
+	}
+
+	@Test
+	public void testTextFieldFeatures() {
+		SeleniumWComponentsWebDriver driver = getDriver();
+
+		SeleniumWCheckBoxWebElement checkBox = driver.findWCheckBox(new ByLabel("this is a checkbox", false));
+
+		assertEquals(false, checkBox.isChecked());
+		checkBox.click();
+		assertEquals(true, checkBox.isChecked());
+	}
+}
```

Comment:
Needs to test getting checked-on-render, disabled (checked and unchecked - there's a challenge!) and read-only (check and unchecked) states too.

I would also suggest tests to ensure click on disabled and read-only do not change their state (if we are being pedantic).